core/tests/webdriverio_desktop/profileFeatures.js

```diff
@@ -185,9 +185,6 @@ describe('Playing the exploration', function() {
   var TEST_USERNAME = 'testUser';
   var TEST_EMAIL = TEST_USERNAME + '@example.com';
 
-  var backButton = element(by.css('.protractor-test-back-button'));
-  var nextButton = element(by.css('.protractor-test-next-button'));
-
```

Comment:
Reverted.

Reply:
Please note: Declaring them directly as variable at top was failing the test:
![Screenshot from 2022-07-10 15-24-52](https://user-images.githubusercontent.com/86483059/178142946-b9cf2973-8734-4881-8e73-5142b47ac736.png)

So, I used functions, in the same way we are using for `$$`.

Reply:
But why is this happening? This is not a all selector right? I think we need to debug it properly because it lead to more flakes in the future.

Reply:
The $ and $$ command is a short way to call the [findElement](https://webdriver.io/docs/api/webdriver#findelement) command in order to fetch a single element on the page and it is not define any where outside the it block in suite files. Also just to be clear we can also use all selectors at top in utils files like we are using single selector but the reason we are using function is because we need to use `await` with all selectors.

Reply:
Thanks for explanation. I understood. But, one question why we can't use $ selectors at the top in case of suites. I know it's giving an error. But the end goal of us should be finding the reason why we can't use it rather than using this alternative because the other alternative was creating failures. It would help both of us if some unnecessary failures came to use because of this. Does webdriverio's documentation contains this info or has any issue created for in webdriverio's repo?

Reply:
We can't use it at the top because it is undefined everywhere in test suites other than the 'it' blocks as I already said.